lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -981,7 +1000,11 @@ def payment_details(payment_method, options = {})
         when String
           token_type_and_details(payment_method)
         else
-          type = network_token?(payment_method) || options[:wallet_type] == :google_pay ? :network_token : :credit
+          type = if network_token?(payment_method)
+                   payment_method.payment_data ? :encrypted_wallet : :network_token
+                 else
+                   options[:wallet_type] == :google_pay ? :network_token : :credit
+                 end
```

Comment:
Do payment methods with options[:wallet_type] == :google_pay have payment_data? would this work?
```suggestion
          type = if network_token?(payment_method) || options[:wallet_type] == :google_pay 
                   payment_method.payment_data ? :encrypted_wallet : :network_token
                 else
                   :credit
                 end
```

Reply:
It doesn't. options[:wallet_type] == :google_pay is for a non-tokenized GooglePay which we treat as a CreditCard